app/session.go

```diff
@@ -369,9 +372,14 @@ func (a *App) CreateUserAccessToken(token *model.UserAccessToken) (*model.UserAc
 
 	token.Token = model.NewId()
 
-	token, err = a.Srv().Store.UserAccessToken().Save(token)
-	if err != nil {
-		return nil, err
+	token, nErr := a.Srv().Store.UserAccessToken().Save(token)
+	if nErr != nil {
+		switch {
+		case errors.As(nErr, &err):
```

Comment:
I would prefer to declare a separate AppError instance and return that. This code is a bit fragile and liable to break.